File: 4229-TrimTrailingVowels/4229-TrimTrailingVowels.py
# Last updated: 4/3/2026, 1:48:31 PM
class Solution:
    def trimTrailingVowels(self, s: str) -> str:
        n = len(s)
        found_non_vowels = False
        vowels = ["a", "e", "i", "o", "u"]
        for i in range(n - 1, -1, -1):
            if s[i] not in vowels:
                found_non_vowels = True
                break
        if not found_non_vowels:
            return ""
        return s[: i + 1]

File: 4229-TrimTrailingVowels/4229-TrimTrailingVowels.cpp
// Last updated: 4/3/2026, 1:48:27 PM
class Solution {
public:
    string trimTrailingVowels(string s) {
        int n = s.length(), idx = -1;
        for (int i = n - 1; i >= 0; i--) {
            if (s[i] != 'a' && s[i] != 'e' && s[i] != 'i' && s[i] != 'o' &&
                s[i] != 'u') {
                return s.substr(0, i + 1);
            }
        }
        return "";
    }
};